### ISSUE TITLE
Support DNS-based GKE Endpoint

### ISSUE THREAD

**Issue by paulwilljones** [google-github-actions/get-gke-credentials#316]
### TL;DR

GKE provides a DNS-based endpoint option when getting cluster credentials that allow for external access to public and private endpoints. Giving the option to use the DNS-based endpoint would allow users to access clusters from GHA using the cluster's FQDN.

### Detailed design

[DNS-based endpoints](https://cloud.google.com/kubernetes-engine/docs/concepts/network-isolation#dns-based_endpoint) for GKE creates an FQDN for the cluster endpoint which is returned for [`gcloud container clusters get-credentials --dns-endpoint`](https://cloud.google.com/sdk/gcloud/reference/container/clusters/get-credentials#--dns-endpoint). This is then used as the endpoint in `kubeconfig` that routes requests via Google Cloud to the public/private cluster endpoint.
Allowing for dns-endpoint to be used would be similar to how `--internal-ip` can be set.

**Comment by sethvargo** [comment]
Released in 2.3.0
